PluralKit.Bot/Commands/Message.cs

```diff
@@ -21,24 +23,98 @@ public class ProxiedMessage
 {
     private static readonly Duration EditTimeout = Duration.FromMinutes(10);
 
-    // private readonly IDiscordCache _cache;
+    private readonly IDiscordCache _cache;
+    private readonly ModelRepository _repo;
     private readonly IClock _clock;
+    private readonly IMetrics _metrics;
 
     private readonly EmbedService _embeds;
     private readonly LogChannelService _logChannel;
     private readonly DiscordApiClient _rest;
     private readonly WebhookExecutorService _webhookExecutor;
 
     public ProxiedMessage(EmbedService embeds, IClock clock,
-                          DiscordApiClient rest,
+                          DiscordApiClient rest, IMetrics metrics, ModelRepository repo,
                           WebhookExecutorService webhookExecutor, LogChannelService logChannel, IDiscordCache cache)
     {
         _embeds = embeds;
         _clock = clock;
         _rest = rest;
         _webhookExecutor = webhookExecutor;
+        _repo = repo;
         _logChannel = logChannel;
-        // _cache = cache;
+        _cache = cache;
+        _metrics = metrics;
+    }
+
+    public async Task ReproxyMessage(Context ctx)
```

Comment:
make sure this method makes all the necessary permission checks that need to happen when re-proxying a message from DMs / another server. it's possible someone might try to reproxy a message after having send message permissions removed